lib/kafka-consumer.js

```diff
@@ -106,7 +106,12 @@ Consumer.prototype._onWrapper = function (consumerInstance, eventName, cb) {
       log.warn('_onWrapper() :: Warning, consumer did not find topic on SR:',
         message.topic);
 
-      message.parsed = JSON.parse(message.value.toString('utf-8'));
+      try {
+        message.parsed = JSON.parse(message.value.toString('utf-8'));
+      } catch (ex) {
+        log.warn('_onWrapper() :: Error parsing value:', message.value,
+          'Error:', ex);
+      }
```

Comment:
is our intention to swallow the error vs throwing it again?

Reply:
hmmm for now its a safe way to deal with that problem. Do you have other suggestion?

Reply:
The idea here was that since we're on Avro, a malformed message that would cause the `JSON.parse` to throw was impossible. 

If it is indeed possible, through user feedback, then we have to somehow notify the consumer app, introduce a global error handler where we can push all errors like that along with the bogus message for debugging?

People will lose their minds if the lib silently eats messages, logging is not enough to qualify this as "non-silent".

The real question here is "do we need this?" have JSON parsing errors been reported?

Reply:
I got your point. The exception is caused only when a avro message with magic byte is in the topic and that schema registry is not in memory. This crashes de application.

My intention is a way to deal with this scenario e the application keeps working, but what you talk about makes sense.

What do you think? Let's put back the exception? In this case a simple `pm2` or docker with restart policy will "fix" the problem

Reply:
Your call, I'm just raising the concerns :P

Reply:
@thanpolas you are right... let's keep the existing "crash" behavior - since this is a patch version. The safer way to fetch will be done in the 2.0 (that I want to start the developing today).